jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java

```diff
@@ -190,9 +188,7 @@ public AbstractConnector(
         if (pool == null)
             pool = _server.getBean(ByteBufferPool.class);
         _byteBufferPool = pool != null ? pool : new ArrayByteBufferPool();
-        addBean(_byteBufferPool);
-        RetainableByteBufferPool retainableByteBufferPool = _server.getBean(RetainableByteBufferPool.class);
-        addBean(retainableByteBufferPool == null ? new ArrayRetainableByteBufferPool() : retainableByteBufferPool, retainableByteBufferPool == null);
+        addBean(_byteBufferPool, pool == null);
```

Comment:
If you do not "forward" the `RetainableByteBuffer` bean for the server to the connector, you're changing the current behavior as all the `findOrAdapt()` use the connector as the container to look into.

You're also breaking the principle of least surprise as you can override the default non-retainable pool by adding a bean on the server but you can't for the retainable pool.

Reply:
I've removed `findOrAdapt`, so a connector will always get it's retainable buffer pool from it's primary buffer pool.

However, I'm a little cautious about how we create buffer pools in each connector.  It might be better to always create a single pool in the server whenever a connector is created without a pool being passed.

Reply:
Right, the existing behavior is slightly broken as it creates one RBBP per connector by default. It should have registered it as a bean onto the server instead of just onto the connector.

But if a user registers a BBP or a RBBP onto the server, those pools are used and shared across all connectors, so you're breaking backward compatibility by not looking for a RBBP bean in the server.